core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java

```diff
@@ -0,0 +1,61 @@
+package org.infinispan.registry.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.registry.InternalCacheRegistry;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * InternalCacheRegistryImpl.
+ *
+ * @author Tristan Tarrant
+ * @since 7.2
+ */
+public class InternalCacheRegistryImpl implements InternalCacheRegistry {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private EmbeddedCacheManager cacheManager;
+   private final Set<String> internalCaches = new HashSet<>();
```

Comment:
Yes, better stay on the safe side. I guess I can just use a ConcurrentHashMap<String, Boolean> where the value is meaningless